### PR TITLE
Broken checkboxes in `MultipleChooserPanel` modal when `USE_THOUSAND_SEPARATOR` is `True`

### PR DESCRIPTION
**Pull request by codesmith25103**
Fixes #...
Issue Number #11134
I attempt to solve the issue as it was mentioned in PR. I have used the filter "|unlocalize" at a correct place to solve this issue

_Please check the following:_

-   [x] Do the tests still pass?[^1]
-   [x] Does the code comply with the style guide?
    -   [x] Run `make lint` from the Wagtail root.
-   [ ] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [x] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [ ] **Please list the exact browser and operating system versions you tested**:
    -   [ ] **Please list which assistive technologies [^3] you tested**:
-   [ ] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

File: wagtail/admin/templates/wagtailadmin/generic/chooser/checkbox_select_cell.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {% load i18n %}
 <td {% if column.classname %}class="{{ column.classname }}"{% endif %}>
-    <input type="checkbox" id="chooser-modal-select-{{ value }}" data-multiple-choice-select name="id" value="{{ value }}" title="{% blocktrans trimmed with title=instance %}Select {{ title }}{% endblocktrans %}">
+    <input type="checkbox" id="chooser-modal-select-{{ value }}" data-multiple-choice-select name="id" value="{{ value|unlocalize }}" title="{% blocktrans trimmed with title=instance %}Select {{ title }}{% endblocktrans %}">
```
Comment (gasman) [inline wagtail/admin/templates/wagtailadmin/generic/chooser/checkbox_select_cell.html:3]: The other occurrence of `value` (in `"chooser-modal-select-{{ value }}"`) needs updating too. Have also just noticed that there's another template that needs the same fixes - `wagtail/admin/templates/wagtailadmin/chooser/tables/page_checkbox_select_cell.html`

**Review by gasman — CHANGES_REQUESTED** [review]

File: wagtail/admin/templates/wagtailadmin/generic/chooser/checkbox_select_cell.html (hunk 1 above)
Comment (codesmith25103) [reply wagtail/admin/templates/wagtailadmin/generic/chooser/checkbox_select_cell.html:3]: Thanks for reviewing. I will make required changes.

**Comment by codesmith25103** [comment]
I have made required changes. Please review it

**Comment by lb-** [comment]
It might be good to have a unit test supporting this also, so that it doesn't break again in the future.

**Review by gasman — CHANGES_REQUESTED** [review]
This is currently failing when I test it on the bakerydemo site (by editing any blog page, and clicking the "Add author" button which should open the chooser popup) because the template isn't loading the `l10n` tag library (which is required for the `unlocalize` filter).

I would have expected the existing unit tests to catch this error, so it seems that we're lacking some test coverage for this.

**Comment by codesmith25103** [comment]
So do I have to write unit test to catch this error?

**Comment by codesmith25103** [comment]
I have added l10n tag in both the files. Hope it will resolve the issue.

**Comment by lb-** [comment]
@codesmith25103  you may want to add a test for this change.

- wagtail/admin/tests/test_edit_handlers.py - see `TestMultipleChooserPanel` - this is more for the panel itself, probably not the view that responds with the checkboxes.
- For the multiple chooser response view, I cannot see that we have a test for this view specifically. It probably needs to go here `wagtail/admin/tests/test_page_chooser.py`. In the class `TestChooserBrowse` a new test that requests with `multiple=1` should get you started.


The view is located in `wagtail/admin/views/chooser.py` (see `BrowseView`) and the URL is set up in `wagtail/admin/urls/__init__.py`.

```py
class TestChooserBrowse(WagtailTestUtils, TestCase):

    @override_settings(USE_THOUSAND_SEPARATOR=True)
    def test_multiple_chooser_view(self):
        response = self.get({"multiple": "1"})  # something like this
        self.assertEqual(response.status_code, 200)
        self.assertTemplateUsed(response, "wagtailadmin/chooser/browse.html")
        # check that the values are correct
        # We may need to add a page that has an id of over 1000
        # we will also need to test this with the setting turned off
        # test data is set in wagtail/test/testapp/fixtures/test.json but we can also add a page on the fly in this test
```

Hopefully that helps as a starting point, a similar test will be needed for the generic multiple chooser but maybe get the page one working first.

**Comment by lb-** [comment]
@codesmith25103 are you OK to create a unit test for this?

**Comment by codesmith25103** [comment]
Hey @lb-, I apologize for the delay. Due to some health and family issues, I wasn't able to complete it. However, I have already started working on it and will update you soon.

**Comment by lb-** [comment]
Thanks @codesmith25103  looks like a new PR has taken this and added unit tests.
https://github.com/wagtail/wagtail/pull/11490

We will ensure we mention you in the release notes, thanks for getting this started.

**Pull request closed without merging**